cmd/flightctl-api/main.go

```diff
@@ -250,14 +263,50 @@ func main() {
 			}
 		}
 
+		// Start Metrics server
 		go func() {
+			log.Info("Starting Metrics server")
 			metricsServer := instrumentation.NewMetricsServer(log, cfg, collectors...)
 			if err := metricsServer.Run(ctx); err != nil {
-				log.Fatalf("Error running server: %s", err)
+				log.Errorf("Metrics server error: %v", err)
+				serverErrors <- err
 			}
-			cancel()
 		}()
 	}
 
-	<-ctx.Done()
+	// Wait for shutdown signal or server error
+	log.Info("All servers started, waiting for shutdown signal...")
```

Comment:
Technically we are waiting for the ctx to be done but is that information necessary for user?